storage/inmemory: panic on delete

## What happened?

The cache server appears to have panic'd when refreshing a user. 

## What did you expect to happen?

:) 

## How'd it happen?

I do not believe Pomerium was under active use when this happened this was on my dev environment. 

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version`): `v0.10.0-dirty+7fd7b3f`


## What did you see in the logs?

```logs
{"level":"info","service":"identity_manager","time":"2020-11-24T14:56:49-08:00","message":"refreshing directory users"}
{"level":"info","service":"identity_manager","user_id":"google/111432655977273150308","time":"2020-11-24T14:56:57-08:00","message":"refreshing user"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x1d44c66]

goroutine 101 [running]:
github.com/pomerium/pomerium/pkg/storage/inmemory.(*DB).ClearDeleted(0xc00064c380, 0x24afd20, 0xc0000440d8, 0xbfe77eac8aa134e8, 0x2051276295b, 0x2f01300)
	/Users/bdd/pomerium/pkg/storage/inmemory/inmemory.go:80 +0x166
github.com/pomerium/pomerium/internal/databroker.New.func1(0xc000935220)
	/Users/bdd/pomerium/internal/databroker/server.go:86 +0x3a3
created by github.com/pomerium/pomerium/internal/databroker.New
	/Users/bdd/pomerium/internal/databroker/server.go:65 +0x2aa```
```

--- a/pkg/storage/inmemory/inmemory.go
+++ b/pkg/storage/inmemory/inmemory.go
@@ -30,7 +30,7 @@ type byIDRecord struct {
 }
 
 func (k byIDRecord) Less(than btree.Item) bool {
-	return k.Id < than.(byIDRecord).Id
+	return k.GetId() < than.(byIDRecord).GetId()
 }
 
 type byVersionRecord struct {
@@ -38,7 +38,7 @@ type byVersionRecord struct {
 }
 
 func (k byVersionRecord) Less(than btree.Item) bool {
-	return k.Version < than.(byVersionRecord).Version
+	return k.GetVersion() < than.(byVersionRecord).GetVersion()
 }
 
 // DB is an in-memory database of records using b-trees.
@@ -77,7 +77,7 @@ func (db *DB) ClearDeleted(_ context.Context, cutoff time.Time) {
 	var remaining []string
 	for _, id := range db.deletedIDs {
 		record, _ := db.byID.Get(byIDRecord{Record: &databroker.Record{Id: id}}).(byIDRecord)
-		ts, _ := ptypes.Timestamp(record.DeletedAt)
+		ts := record.GetDeletedAt().AsTime()
 		if ts.Before(cutoff) {
 			db.byID.Delete(record)
 			db.byVersion.Delete(byVersionRecord(record))

--- a/pkg/storage/inmemory/inmemory_test.go
+++ b/pkg/storage/inmemory/inmemory_test.go
@@ -74,4 +74,14 @@ func TestDB(t *testing.T) {
 		require.NoError(t, err)
 		assert.Len(t, records, 0)
 	})
+	t.Run("delete twice", func(t *testing.T) {
+		data := new(anypb.Any)
+		assert.NoError(t, db.Put(ctx, "abcd", data))
+		assert.NoError(t, db.Delete(ctx, "abcd"))
+		assert.NoError(t, db.Delete(ctx, "abcd"))
+		db.ClearDeleted(ctx, time.Now().Add(time.Second))
+		record, err := db.Get(ctx, "abcd")
+		require.Error(t, err)
+		assert.Nil(t, record)
+	})
 }

